### ISSUE TITLE
Not Working

### ISSUE THREAD

**Issue by SilverMax-82** [netsampler/goflow2#151]
Hello!
I deployed [Kafka+Clickhouse+Grafana](https://github.com/netsampler/goflow2/blob/main/compose/kcg) Docker.
Config netflow on my Cisco router. But i don't see any graphs on grafana.
Please tell me what should I check and how?

Cisco config:
```
flow record NetFlow-Record
 match ipv4 ttl
 match ipv4 tos
 match ipv4 protocol
 match ipv4 source address
 match ipv4 destination address
 match transport source-port
 match transport destination-port
 match interface input
 collect interface output
 collect counter bytes long
 collect counter packets long
 collect timestamp absolute first
 collect timestamp absolute last
 collect application name
!
!
flow exporter goflow
 description Export to goflow
 destination 10.77.77.202
 source Port-channel2.10
 transport udp 2055
 template data timeout 30
!
!
flow monitor NetFlow-Monitor-input
 exporter goflow
 cache timeout active 60
 record NetFlow-Record
!
interface Tunnel10
 ip flow monitor NetFlow-Monitor-input input

```
![image](https://user-images.githubusercontent.com/85797365/228545995-153b77ce-7fe8-49e3-9fc4-974d10702763.png)

![image](https://user-images.githubusercontent.com/85797365/228547285-dd8c3f88-042d-4ed3-b513-e5aa400caaa3.png)

**Comment by lspgn** [comment]
Hello,
It would be hard to say without a packet capture.
Since it's NetFlow: maybe it needs to have the template?
To make sure, could you plot (either in Grafana or via Prometheus which should be accessible via http://localhost:9090) the following timeseries `flow_process_nf_errors_count`?

**Comment by SilverMax-82** [comment]
Hello, thank you for answer.
I'm sorry - a don't know how work with both systems (grafana and prometheus) :)
I don't know how to make these requests.
I tried in prometheus:

![image](https://user-images.githubusercontent.com/85797365/228774148-62b4c147-fc03-4b89-81c8-f950f2d3ccf6.png)

**Comment by SilverMax-82** [comment]
oh, sorry, was incorrect screenshot:

![image](https://user-images.githubusercontent.com/85797365/228775386-01536f0e-729e-470f-a3f0-f64972381b13.png)

**Comment by lspgn** [comment]
@SilverMax-82
Perfect, could you do the same with `flow_process_nf_flowset_records_sum`? In `graph` mode.
This will tell me if GoFlow2 is seeing actual data samples. It has templates so far.

Alternatively, you can run a `goflow2` binary directly and observe the JSON output.
If you do not see traffic, could you provide me a packet capture?

One more thing to check: can you send me logs of the `docker-compose` command? 
```bash
docker ps
```
This will tell me if something broke between GoFlow2 and ClickHouse.
As a side note: it might be hard to debug the whole stack since it's not the core project. You may want to have a look at [Akvorado](https://github.com/akvorado/akvorado) for a more bundled up solution (that uses exact same software under the hood).

**Comment by SilverMax-82** [comment]
```
docker ps 
CONTAINER ID   IMAGE                                            COMMAND                  CREATED      STATUS      PORTS                                                                                                                             NAMES
fc01aa377062   netsampler/goflow2                               "./goflow2 -transpor…"   5 days ago   Up 5 days   0.0.0.0:2055->2055/udp, :::2055->2055/udp, 0.0.0.0:6343->6343/udp, :::6343->6343/udp, 0.0.0.0:8080->8080/tcp, :::8080->8080/tcp   kcg_goflow2_1
07b56b7c30f0   clickhouse/clickhouse-server:22.8.14.53-alpine   "/entrypoint.sh"         5 days ago   Up 5 days   9000/tcp, 0.0.0.0:8123->8123/tcp, :::8123->8123/tcp, 9009/tcp                                                                     kcg_db_1
97d6e6ece607   bitnami/kafka:3.4.0                              "/opt/bitnami/script…"   5 days ago   Up 5 days   0.0.0.0:9092->9092/tcp, :::9092->9092/tcp                                                                                         kcg_kafka_1
7ec91add559e   grafana/grafana:9.4.3                            "/run.sh"                5 days ago   Up 5 days   0.0.0.0:3000->3000/tcp, :::3000->3000/tcp                                                                                         kcg_grafana_1
73dfaa7db51b   prom/prometheus:v2.37.6                          "/bin/prometheus --c…"   5 days ago   Up 5 days   0.0.0.0:9090->9090/tcp, :::9090->9090/tcp                                                                                         kcg_prometheus_1
2f7e8ec52fc5   bitnami/zookeeper:3.7.1                          "/opt/bitnami/script…"   5 days ago   Up 5 days   2888/tcp, 3888/tcp, 0.0.0.0:2181->2181/tcp, :::2181->2181/tcp, 8080/tcp                                                           kcg_zookeeper_1

```

**Comment by lspgn** [comment]
@SilverMax-82 thanks!

I am not seeing anything out of the ordinary and you should see data inside ClickHouse :( 

Could you try the following command:
```bash
curl -XPOST --data 'SELECT count() FROM flows_raw' http://localhost:8123
```

**Comment by SilverMax-82** [comment]
Yes, i think problem with clickhouse. But i don't know how to debug it.
Also maybe incorrect query in grafana.
```
curl -XPOST --data 'SELECT count() FROM flows_raw' http://localhost:8123
56433314

```

**Comment by SilverMax-82** [comment]
From grafana:
```
SELECT
    if(etype = 0x800, IPv4NumToString(reinterpretAsUInt32(substring(reverse(src_addr), 13,4))), IPv6NumToString(src_addr)) as srcip,
    sum(bytes*sampling_rate) AS sumbytes
FROM $table
WHERE $timeFilter

```
Maybe incorrect SELECT?

**Comment by lspgn** [comment]
So the data is making its way to ClickHouse.

What's the result of:
```
curl -XPOST --data 'SELECT sampling_rate,count() FROM flows_raw GROUP BY sampling_rate' http://localhost:8123
```

**Comment by lspgn** [comment]
Also inside the Grafana query, could you click on `Generated SQL` and run it with the cURL like above?
I'd be curious to see if there is no data or it's just improperly counted.

**Comment by SilverMax-82** [comment]
```
curl -XPOST --data 'SELECT sampling_rate,count() FROM flows_raw GROUP BY sampling_rate' http://localhost:8123
0	56541584

```

![image](https://user-images.githubusercontent.com/85797365/229126713-ef5db94b-0983-4ed4-8d11-03627731e807.png)

**Comment by lspgn** [comment]
What model of Cisco is this?
It's not passing the sampling rate (via OptionDataTemplate). As a result, any dashboard plotting the amount of bytes and packets that multiply by sampling_rate results to being zero.
I would need a capture of the template. It might be similar issue to #113.
Or it could be a router configuration issue.

There is a way of forcing the sampling rate via the enricher but setting up is less trivial (see #21).
Another way of solving it would be to hardcode (or use a dictionary) inside the ClickHouse queries: replacing `sampling_rate` by the actual number.

**Comment by SilverMax-82** [comment]
I use Cisco ISR4431
Can you write more detailed - what i must do?
Where apply the -samplingrate=1024 ?
or what the actual number of sampling rate? :)

**Comment by lspgn** [comment]
So [it appears](https://community.cisco.com/t5/routing/how-to-define-sampling-rate-in-netflow-of-cisco-isr4431/td-p/4088412) by default the sampling is 1:1.

You have multiple solutions:
* Try setting up sampling per the doc and hopefully the device will send the sampling information. This may be better in the future to reduce the load on the database.
* You can change the SQL queries in the Grafana panels and remove `sampling_rate`. This is the quickest solution but may be difficult if you have a large diversity of network devices sending flows to keep consistency.
* Pipe to an [enricher](https://github.com/netsampler/goflow2/blob/main/cmd/enricher/main.go) that is configured to overwrite the `sampling_rate` field before being inserted inside the database, unfortunately I do not have a guide on this.

**Comment by gaby** [comment]
@lspgn I'm having this exact issue. I'm sending data using `softflowd`. When using a sampling rate of 0 which would get everything the graph is flat. I tried changing the sample rate and data doesnt show up even though running the queries above return results.

The only way I was able to get data in was using v5 with the legacy listener. When doing that the `goflow internal` dashboard would remain empty.

My cmd is something like: `softflowd -d -s 0 -A mili -i eth0 -v 9`
